Add missing database entries.

NullCGT opened the pull request:
The goal of this PR is to add entries to the database for all items and monsters that currently lack entries.

Entries that were missing before this PR was opened: https://bilious.alt.org/~paxed/nhdev/dataref.txt
Script to determine which entries are missing: https://bilious.alt.org/~paxed/nhdev/dataref-370.pl.txt

The entries I've added thus far are suggestions. I've tried to shoot for literary quotes that do not duplicate existing database entries. The current quotes I've suggested are fairly biased towards the western literary canon, so I'm especially interested in finding quotes from a more diverse set of sources to fill out the database with.

Commit:
Add missing database entries.

This commit adds entries for the following items:
    - Credit card
    - Silver bell
    - Bell
    - Walking shoes
    - Iron shoes
    - Hard shoes
    - Dragonbane
    - Gauntlets of Power
    - Gauntlets of Fumbling
    - Gauntlets of Dexterity
    - Potion of Sickness
    - Splashes of Venom (Various)
    - Shields (Various)

The following monsters also now have associated entries:
    - Stalker

entrez commented:
I like this, but the stalker quote doesn't make sense to me.  It seems like it was just selected because the character is called a stalker, even though it's unrelated to the NetHack stalker.  You can't use it to learn about or better imagine invisible stalkers.

The Wilde bon mot for credit card also feels trite to me.

NullCGT commented:
> I like this, but the stalker quote doesn't make sense to me. It seems like it was just selected because the character is called a stalker, even though it's unrelated to the NetHack stalker. You can't use it to learn about or better imagine invisible stalkers.
> 
> The Wilde bon mot for credit card also feels trite to me.

Fair enough. I was struggling to find a quote that fit "credit card" that did not feel trite in context.
As for the Stalker one, I'll admit that I chose it for the novelty of using a quote from a film of the same name about something that one character does not see or recognize. I'm sure there's something better out there.

Commit:
Encyclopedia Entry: Ice Box.

copperwater commented:
See also [xNetHack data.base](https://github.com/copperwater/xNetHack/blob/master/dat/data.base) which has added some of these same missing ones: 
- bell is the first stanza of Edgar Allan Poe's "The Bells" (the quote you used is also good, but the poem is terrific)
- credit card uses a passage from the Hitchhiker's Guide series about credit cards and their uselessness in fantastic settings
- invisible stalker uses a passage from H. G. Wells' "The Invisible Man"

It also adds quotations for conducts and various other things: 
- alchemy
- athame
- atheist
- Bell of Opening (moves the existing quote to Book of the Dead, which previously shared an entry with the Candelabrum)
- blind [from birth]
- candle
- cloak of invisibility
- deaf [from birth]
- dented pot
- dwarvish cloak
- foodless / went without food
- fortune cookie
- fruit
- hallucination
- illiterate
- iron
- leather
- Moloch (a quotation already exists, a second is added here)
- nudist
- pacifist
- scroll of punishment
- shade
- silver
- vegan / vegetarian
- wand (adds a second quotation)
- weaponless

(entrez found a lot of these, so credit to him)

NullCGT commented:
@copperwater Those are excellent; I would love to pull those into this PR / see another PR with those.

Also, I read the first stanza of the Bells, and you're totally right. That's an absolutely terrific quotation to include.

Commit:
Encyclopedia: Bells, credit cards, and stalkers.

Replace the encyclopedia entries for bells, credit cards, and
stalkers with those suggested by aosdict. Sourced from xnethack.

Commit:
Rectify typo in shield database entry.

Commit:
Return strange object entry for strange and glorkum.

If the player is looking up a glorkum in the encyclopedia then
there are probably larger problems at play than a missing encyclopedia
entry.

copperwater commented on dat/data.base:
```diff
@@ -503,18 +503,23 @@ bell of opening
-	melted down their mugs; and they now hung, nameless and mugless, 
-	in the church-tower.
-		[ The Chimes, by Charles Dickens ]
+~*engraved*bell*
```
Vanilla doesn't have "engraved silver bell" as the unidentified description for Bell of Opening so this line isn't needed.

copperwater reviewed (commented):
Feel free to add the other entries I listed in this PR; I don't see the need of making a separate one for it.

Commit:
Remove unneeded engraved bell encyclopedia entry.

Commit:
Add fortune cookie quotation.

Sourced from xnethack.

Commit:
Add athame quotation.

Sourced from xnethack.

Commit:
Added dented pot encyclopedia entry.

Sourced from xnethack.

Commit:
Add dwarvish cloak encyclopedia entry.

Sourced from xnethack.

Commit:
Add fortune cookie encyclopedia entry.

Sourced from xnethack.

Commit:
Add fruit/fruitname encyclopedia entry.

Sourced from xnethack.

Commit:
Add hallucination encylopedia entry.

Sourced from xnethack.

Commit:
Ensure potion of hallucination matches encyclopedia.

Commit:
Scroll of punishment encyclopedia entry.

Sourced from xnethack. Added a line so that it will match "punished"
as well as "punishing."

Commit:
Replace shade encyclopedia entry.

Sourced from xnethack. Seems like an excellent candidate for
replacement, since the quote gets across what shades are and also
hints at their weakness.

Commit:
Add Demonbane encyclopedia entry.

Commit:
Add database entry for white-handed shield.

Commit:
Match "large rock" for rock encyclopedia entry.

Commit:
Add encyclopedia entry for pointy hats.

Commit:
Looking up custom fruitnames in the encyclopedia.

Looking up a custom fruitname in the encyclopedia will now yield
the encyclopedia entry for "fruit" if another matching database
entry is not found. This preserves the fun associated with naming
fruits after existing objects, while also preventing a failure to
find.

Commit:
Add encyclopedia entry for helm of brilliance.

entrez commented on src/pager.c:
```diff
@@ -831,14 +831,20 @@ checkfile(char *inp, struct permonst *pm, boolean user_typed_name,
+           then "fruit" is the alternate description. We do this here so that
+           if the fruit name is an extant object, looking at the fruit yields
+           that object's description. */
+        if (!alt && !strncmpi(dbase_str, gp.pl_fruit, PL_FSIZ))
```
If you want to match fruitnames you might want to check `fruit_from_name(dbase_str, TRUE, (int *) 0) != (struct fruit *) 0` so it will include fruits that still exist in-game after fruitname has been changed by the player.

entrez commented on dat/data.base:
```diff
@@ -2234,6 +2363,11 @@ hell hound*
 	dark form and savage face which broke upon us out of the wall
 	of fog.
 	  [ The Hound of the Baskervilles, by Sir Arthur Conan Doyle ]
+helm of brilliance
```
I don't understand what about the quote is specific to a helm of brilliance, unless it's "brilliance" in the sense of "radiant beauty" (which I don't think is how it's used in the item name).  This seems to me like it would be a better quote for "helmet" in general.

entrez commented on dat/data.base:
```diff
@@ -1234,6 +1293,23 @@ major demon
 	them difficult to deal with, even on the rare occasions when
 	they are friendly.
 	  [ The Tough Guide to Fantasyland, by Diana Wynne Jones ]
+demonbane
```
This seems like another "has no bearing on the item/monster other than an abstract association with its name" one to me.  The apologetics is not totally unjustified since it's the sac gift for Priests, so I guess that is a secondary link, but there must be other options in that field that are actually referring to a (metaphorical) weapon driving out demons rather than "the continued faith of a man confronted with a seemingly godless universe".  E.g. Matthew 10:34 could be appropriate, if maybe a little obvious.

entrez commented on dat/data.base:
```diff
@@ -2470,6 +2604,16 @@ hunter
 	Where is the haste that ye hurry by?
 	Brother, I go to my lair to die.
 		[ The Jungle Book, by Rudyard Kipling ]
+ice box
```
I would propose This Is Just To Say by William Carlos Williams for this :-)
